### DISCUSSION TITLE
Nested inline block

### DISCUSSION THREAD

**Discussion by ivandokov** [codex-team/editor.js#2255] [Q&A]
I am building something like Wordpress'es shortcodes functionality where a piece of text is replaced with data from the database.
I want to be able to insert visual elements (imagine a span with some styling - `Hello <span class="...">user-full-name</span>`) for each shortcode instead of using text format (e.g. `Hello [user-full-name].`). And I want to offer a list of available shortcodes to the end user.

I can't seem to find such functionality in Editor.js and I thought I might ask. Is there a way to have nested blocks?
Also, this brings another challenge - how do I open the inline toolbar without having a text selected first?

**Comment by gkovats** [comment]
Inline toolbar works with HTML tags. You can build custom inline tools around unique tags, possibly even just use custom HTML elements like `<user data-name="">...</user>`, but it won't support WIKI - it's reading DOM tags. 

Believe the API has an `api.inlineToolbar.open()` method to manually trigger opening the method. In your case, you could get by with building an inlineToolbar Tool. Try copying https://github.com/editor-js/link/blob/master/src/index.js as a baseline, and changing it to your own custom tag with a form that supports the attributes / inputs you need.

Not a complete answer, not an expert either. FWIW.
